### PR TITLE
allow user to update linked app asynchronously

### PR DESCRIPTION
**Pull request by mkangia**
https://manage.dimagi.com/default.asp?282955#BugEvent.1528448

This adds ability for a user to initiate linked app update and get notified via email once its done.
Useful for big projects like icds that are not able to update and time out.

View changes:
1. Add a checkbox to update asynchronously instead
2. Successful message in case the request is submitted successfully

screen shot 2018-10-15 at 3 31 48 pm

Email messages can be found [here](https://github.com/dimagi/commcare-hq/pull/21885/files#diff-87bcd4400226fe34d7fdb28010cfe4b2R294)

### CODE REVIEW

**Comment by mkangia** [comment]
@dimagi/product @dimagi/ux 
any thoughts on UI changes or messages used.

**Review by snopoke — APPROVED** [review]
nice.

**Comment by orangejenny** [comment]
Would there be a downside to making the new option (update async & email when done) the only option?

File: corehq/apps/app_manager/views/utils.py
Comment (orangejenny) [inline corehq/apps/app_manager/views/utils.py]: translate subject line?

File: corehq/apps/app_manager/templates/app_manager/partials/settings/app_settings.html
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/settings/app_settings.html]: Since using real booleans doesn't work cleanly here, I think it's more readable to use `notify=1` rather than the string `'true'`.

File: corehq/apps/app_manager/templates/app_manager/partials/settings/app_settings.html
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/settings/app_settings.html]: There should only be one `.btn-primary` on a page. Whichever of these is more common should use it, and the other one should use `.btn-default`.

**Review by orangejenny — CHANGES_REQUESTED** [review]
Minor comments.

**Comment by mkangia** [comment]
@orangejenny 

> Would there be a downside to making the new option (update async & email when done) the only option?

I don't have good context to weigh in on that but i feel it might not be useful for small linked apps(if there are any) since that can happen pretty quickly rather than waiting for a background task to pick it up. No strong opinion about that in particular. Whatever the product team decides.

File: corehq/apps/app_manager/templates/app_manager/partials/settings/app_settings.html
Comment (mkangia) [reply corehq/apps/app_manager/templates/app_manager/partials/settings/app_settings.html]: > Since using real booleans doesn't work cleanly here

Can you share the reason? Ultimately in the view we always would need to string comparison. Or does it look better in urls?

Comment (orangejenny) [reply corehq/apps/app_manager/templates/app_manager/partials/settings/app_settings.html]: Nevermind, I don't think it really matters. I tend to forget some of javascript's rules around string => boolean conversions, but thinking about it some more, I don't think "1" is really any better than "true".

**Comment by orangejenny** [comment]
> No strong opinion about that in particular. Whatever the product team decides.

Cool, I don't feel that strongly, happy to defer to product.

File: corehq/apps/app_manager/templates/app_manager/partials/settings/app_settings.html
Comment (mkangia) [reply corehq/apps/app_manager/templates/app_manager/partials/settings/app_settings.html]: Cool. Waiting on feedback on keeping just one or both buttons from Product and then would change accordingly

**Comment by mkangia** [comment]
@ohmegasquared 
> Is it just to email once it's finished? 

That is correct. For larger applications where it takes time. Once can choose to do this asynchronously.

**Comment by ohmegasquared** [comment]
Thanks! I can image a couple of scenarios:
1) one CTA "Update from Master App" that opens in a new tab to something like "Please wait will your Application is being linked" with a loading or process indicator and a CTA to say "Email me when complete" so it can run in a background tab while the user can choose to use CCHQ or watch it link. 
or
2) Having a check box to select "Email me when finished (recommended for large projects)" and the CTA "Update from Master App"
or 
3) two CTAs "Update" and "Update and email when finished"

**Comment by mkangia** [comment]
thanks @ohmegasquared 
the first suggestion was also shared by Cody and i believe thats going to take quite some breaking at the code.
I am going to try to incorporate the second one.

File: corehq/apps/app_manager/views/utils.py
Comment (esoergel) [inline corehq/apps/app_manager/views/utils.py]: This should be `six.text_type` or something for Python3 compatability

**Review by esoergel — APPROVED** [review]

**Comment by mkangia** [comment]
@ohmegasquared update view to show a checkbox instead.
Also update the description above with new screenshot for reference. Please note the checkbox is not checked by default.

File: corehq/apps/app_manager/views/utils.py
Comment (mkangia) [reply corehq/apps/app_manager/views/utils.py]: thanks, updated.

**Review by esoergel — APPROVED** [review]

**Comment by mkangia** [comment]
@orangejenny any outstanding review here?

**Comment by orangejenny** [comment]
@mkangia nope, looks good to me

**Review by orangejenny — APPROVED** [review]

**Pull request merged**
